rfc/system/0001-flyte-execution-tags.md

```diff
@@ -0,0 +1,127 @@
+# Flyte Execution Tags and Metadata
+
+**Authors:**
+
+- @kumare3
+
+## 1 Executive Summary
+
+Flyte currently provides no visual ways of grouping executions and other
+entities with a “project” and “domain”. Usually a project is used by a team of
+engineers and/or researchers, and sometimes they may be  experimenting in
+isolation or running multiple co-horts of experiments. Also, there are cases in
+which grouping executions by a group might make them easier to find and
+associate them better. This document provides a solution of how we could
+improve the experience of discovering executions within Flyte. It also provides
+motivation of how this feature could further improve other ecosystem projects.
+
+## 2 Motivation
+
+As a User I want to
+ - Group a certain number of executions into an experiment group - for ease of debugging and discovery - when launching the workflows or after they have already finished.
+ - I want to mark certain executions as “blessed” or “released”. This could be done through providing a semantic version after the execution is successful 
+ - I want to group all “sub launchplan” executions with the parent execution.  
+ - External systems could group executions based on some identifiers.  
+ - Users could name their executions without having to worry about the character limits, uniqueness constraints and limited characterset.
+ - Simplify filtering of certain executions
+ - Be able to remove tags from an execution after it has been started, or after it has finished.
+
+## 3 Proposed Implementation
+
+### Support for tags
+
+We propose to solve the problem of discovery by supporting arbitrary metadata association with an entity. This is similar to concept of “tags” as in AWS.
+The tags are represented as plain string.
+We'll add tags to [ExecutionCreateRequest](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#executioncreaterequest)  -> [ExecutionSpec](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/admin/admin.html#executionspec).
+
+The resultant tags will be persisted in the database, instead of being applied to the
+execution in Kubernetes. We'll create two new tables in the flyteadmin database.
+- ``execution_admin_tags`` is a join table, and it contains admin_tags ID and execution ID.
+- ``admin_tags`` saves all the tag names
+
+As a first step, we recommend that these tags are
+persisted associated with an execution and then ListExecutions API is updated
+to return all 
+ - filtered executions by tags with supported `and` `or` queries
+ - all associated executions with every execution
+ - limit total number of tags per execution to 10-15
```

Comment:
Nit-pick (non blocking): this last bullet point appears to not fit "ListExecutions API is updated
to return all".